python/tskit/trees.py

```diff
@@ -3826,12 +3851,14 @@ def max_root_time(self):
         Returns time of the oldest root in any of the trees in this tree sequence.
         This is usually equal to ``np.max(ts.tables.nodes.time)`` but may not be
         since there can be nodes that are not present in any tree. Consistent
-        with the definition of tree roots, if there are no edges in the tree
-        sequence we return the time of the oldest sample.
+        with the definition of tree roots, if there are no edges in the tree sequence
+        we return the time of the oldest sample, or ``None`` if there are no samples.
 
         :return: The maximum time of a root in this tree sequence.
         :rtype: float
         """
+        if self.num_samples == 0 and self.num_edges == 0:
+            return None
```

Comment:
In other places we raise value error for situations like this, so I'm not sure that `None` is the right return. If we're going with it then it should be in the docstring.

Reply:
It is in the docstring (above, I think), but perhaps you are right: if `max([])` returns an error then perhaps ts.max_root_time should too? If so we should document that.